src/transformers/modeling_utils.py

```diff
@@ -729,6 +730,16 @@ def _load_state_dict_into_meta_model(
         elif not is_quantized:
             # For backward compatibility with older versions of `accelerate`
             set_module_tensor_to_device(model, param_name, param_device, **set_module_kwargs)
+        elif param.dtype == torch.uint8:
```

Comment:
for stronger check, you can also add this condition (and just in case some uint8 tensors end up in the state dict of non-quantized models, to be on the safe zone)
```suggestion
        elif param.dtype == torch.uint8 and is_quantized:
```

Reply:
Done. `is_quantized` is asserted now and then code branches into 4 / 8 bits.